### DISCUSSION TITLE
MFXListView set background color

### DISCUSSION THREAD

**Discussion by Lairon1** [palexdev/MaterialFX#333] [Q&A] [answered]
I'm trying to set the background color for an MFXListView, but it doesn't work.
I tried both `-fx-background-color` and `-mfx-track-color` `-mfx-thumb-color` `-mfx-thumb-hover-color`. In all cases, nothing changes at all. I also tried to change the `trackColor` etc in the FXML file itself, which also does not give any results.
help me please

**Comment by infinite-dev22** [comment]
In JavaFX, the background color of the ```ListView``` is set or defined on the cells. So to set a background color for ```MFXListView```, using CSS set styles on CSS class ```mfx-list-cell```.
Using Java, follow the approach below
```
import io.github.palexdev.materialfx.controls.MFXListView;
import io.github.palexdev.materialfx.controls.cell.MFXListCell;
import javafx.application.Application;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.scene.Scene;
import javafx.scene.layout.BorderPane;
import javafx.stage.Stage;

import java.util.List;

public class Main extends Application {
    public static void main(String[] args) {
        launch(args);
    }

    @Override
    public void start(Stage stage) throws Exception {
        List<String> list = List.of("John", "Jamaine", "Jonathan", "Jane");
        
        ObservableList<String> observableList = FXCollections.observableList(list);
        
        MFXListView<String> listView = new MFXListView<>(observableList);
        
        listView.setCellFactory((s) -> {
            MFXListCell<String> cell = new MFXListCell<>(listView, s);
            cell.setStyle("-fx-background-color: red;");
            return cell;
        });

        BorderPane borderPane = new BorderPane(listView);

        stage.setScene(new Scene(borderPane));
        stage.show();
    }
}
```
If this doesn't help, edit your question and provide an MRE so this can be examined much more closely.

### Edit:
Reference [Stack Overflow](https://stackoverflow.com/questions/33454279/listview-styling-in-javafx).

**Reply by Lairon1** [reply]
My cell already has a style that changes its background. But the background changes only in the cell itself, but not for the entire list.
In the production screenshot, all that is white is the MFXListView area. Also, sometimes there are white stripes of celestial bodies between the cells, as I understand it, because the background is white (Screenshot 2).

![image](https://github.com/palexdev/MaterialFX/assets/62989407/6c64a38a-196d-4f44-b207-443e2c0cbb52)
![image](https://github.com/palexdev/MaterialFX/assets/62989407/ba102219-fceb-4f20-9dbe-3afd096d2793)

**Reply by Lairon1** [reply]
**Summarize:** All the white overgrowth in screenshot 1 is an `MFXListView`, I want it to be color `#303030`. Changing the color of a cell only changes the color of the cell

**Reply by infinite-dev22** [reply]
Style using a CSS style sheet, then set styles for ```MFXListCell``` like below:
```
.mfx-list-cell,
.mfx-list-cell:empty {
   // your styles in here.
}
```

**Reply by infinite-dev22** [reply]
Then for the white stripes, I think those are the cells borders. Setting the border color on your cells should fix that.

**Reply by Lairon1** [reply]
My `ManifestVersionCell.css` file:

```
.cellBackground{
    -fx-background-color: #424242;
}

.cellBackground:empty {
    -fx-background-color: #303030;
}

.text{
    -fx-text-fill: white;
    -fx-opacity: 0.7;
    -fx-font-family: "Inter Semi Bold";
    -fx-font-size: 14px;
}
```

This is how I set the style for the cell:
```
<fx:root type="io.github.palexdev.materialfx.controls.cell.MFXListCell" prefHeight="68.0" prefWidth="266.0" styleClass="cellBackground"
         stylesheets="@../../../../../css/ManifestVersionCell.css"
         xmlns="http://javafx.com/javafx/17.0.2-ea" xmlns:fx="http://javafx.com/fxml/1">
    <children>
        <BorderPane>
           ...
        </BorderPane>
    </children>

</fx:root>
```

And it doesn't work

**Reply by infinite-dev22** [reply, marked as answer]
Ok then, set the ```mfx-list-cell``` background color to transparent. Then set the ```virtual-flow``` background color to transparent too(use ```.mfx-list-view .virtual-flow```). And finally set ```mfx-list-view``` background color to your choice.

```
.mfx-list-cell {
   -fx-background-color: transparent;
   -fx-border-color: transparent;
}

.mfx-list-view .virtual-flow {
    -fx-background-color: transparent;
}

.mfx-list-view {
    -fx-background-color: red;
}
```

**Reply by Lairon1** [reply]
It worked! Thank you

**Comment by infinite-dev22** [comment]
Still the styling of the MFXListView is done on it's cells. There are many psuedoclasses assigned to ```MFXListCell```, these can be used to style it in various ways possible. Some of the psuedoclasses assigned to style class ```mfx-list-cell``` are:
```
:hover
:selected
:empty
```

You can reference the OpenJFX documentation about pseudoclasses on ListCell, am sure those will be able work on MFXListCell too.
